api/endpoints/skill.py

```diff
@@ -215,6 +216,39 @@ async def list_sub_skills(*, root_skill_id: str, user: User | None = public_auth
     )
 
 
+@router.get("/skilltree/subskills/search", responses=responses(SearchResults))
+@redis_cached("skills", "search_term", "user")
+async def search_sub_skill(*, search_term: str, user: User | None = public_auth) -> Any:
```

Comment:
This endpoint should not need authentication, so the `user` parameter can be removed here.

```suggestion
@redis_cached("skills", "search_term")
async def search_sub_skill(*, search_term: str) -> Any:
```